lib/components/settings/update_manager.dart

```diff
@@ -190,13 +215,36 @@ abstract class UpdateManager {
 
     OpenFilex.open(file.path);
   }
+
+  static Future<String> getNewestVersionChangelog() async {
+    final int newestVersion;
+    newestVersion = await getNewestVersion() ?? 0;
+    return getChangelogText(newestVersion);
+  }
+
+  static Future<String> getChangelogText(int newestVersion) async {
+    return await downloadChangelogText(newestVersion);
+  }
+
+  static Future<String> downloadChangelogText(int newestVersion) async {
+    String changeLog = '';
+    var downloadUrl =
+        'https://raw.githubusercontent.com/adil192/saber/main/metadata/en-US/changelogs/$newestVersion.txt';
+    var response = await http.get(Uri.parse(downloadUrl));
+    if (response.statusCode == 200) {
```

Comment:
Instead of this, please adopt the approach used in `_downloadLatestVersionFileFromGitHub` where `SocketException`s are thrown